Create STAC1.0

scottyhq opened the pull request:
First pass at updates for new packaging setup and PySTAC1.0 / STAC1.0

Test catalog + stac browser here https://scottyhq.github.io/sentinel1-rtc-stac

scottyhq commented on src/stactools/sentinel1/constants.py:
```diff
@@ -28,6 +28,10 @@
     name='Indigo Ag Inc.',
     roles=['processor', 'licensor'],
     url='https://registry.opendata.aws/sentinel-1-rtc-indigo')
+# https://github.com/stac-utils/pystac/issues/479
+# 'processing:level': 'L3',
+# 'processing:lineage': 'https://sentinel-s1-rtc-indigo-docs.s3-us-west-2.amazonaws.com/methodology.html',  # noqa: E501
+# 'processing:software': {"S1TBX": "7.0.2"}
```
waiting for feedback on https://github.com/stac-utils/pystac/issues/479

scottyhq commented:
@gadomski @matthewhanson I think this is ready for review. Looks like I can't request reviewers and don't have merge rights here

scottyhq commented on src/stactools/sentinel1/rtc_metadata.py:
```diff
@@ -147,11 +149,14 @@ def epsg(self) -> Optional[str]:
+            'mgrs:utm_zone': self.metadata['TILE_ID'][:2],
+            'mgrs:latitude_band': self.metadata['TILE_ID'][2],
+            'mgrs:grid_square': self.metadata['TILE_ID'][3:],
             'sentinel:mgrs': self.metadata['TILE_ID'],
```
keeping this as a property b/c was unsure how to merge multiple `mgrs:properties` when creating a test catalog https://github.com/stac-utils/pystac/issues/480

gadomski commented on README.md:
````diff
@@ -1,15 +1,39 @@
+```
+
+Create a STAC Item
+```
+stac sentinel1 create-item s3://sentinel-s1-rtc-indigo/tiles/RTC/1/IW/12/S/YJ/2016/S1B_20161121_12SYJ_ASC S1B_20161121_12SYJ_ASC
+```
````
Is this missing the destination argument?

scottyhq replied:
it is there (same last subfolder name)

gadomski commented on README.md:
````diff
@@ -1,15 +1,39 @@
+```
+git checkout -b newfeature
+# make changes
+./scripts/lint
+./scripts/format
+./scripts/test
````
I like breaking this apart to show the steps, but could also mention that `scripts/cibuild` will do it all-in-one and can be a nice single-command check before submitting a PR.

gadomski commented on environment.yml:
```diff
@@ -3,4 +3,5 @@ channels:
   - defaults
 dependencies:
-  - gdal
+  - python=3.8
```
Do you need to specify python version in the conda env? We _should_ be able to work w/ multiple Python envs.

scottyhq replied:
right, i removed the python  from this

gadomski commented on environment.yml:
```diff
@@ -3,4 +3,5 @@ channels:
 dependencies:
-  - gdal
+  - python=3.8
+  - rasterio>=1
```
See https://github.com/stactools-packages/aster/pull/4/files for some backflips to get `rasterio` builds working through pip. Per https://github.com/stac-utils/stactools/issues/144, our conda vs. pip story a bit confused. I think for subpackages it's ok to do things a little different, but IMO if the build instructions don't need `conda`, then we should try to keep it just to pip. Did you need a `conda` env to make tests work locally?

scottyhq replied:
i'm just accustomed to using conda for virutal environments... could add alternative docs later for just pip

gadomski commented on setup.cfg:
```diff
@@ -28,7 +28,7 @@ package_dir =
 packages = find_namespace:
 install_requires =
-    stactools ~= 0.1
+    stactools ~= 0.2.1a1
```
```suggestion
    stactools == 0.2.1a1
```

While we're in prerelease mode for stactools, a new prerelease could be a breaking change, so we should hard pin.

gadomski commented on src/stactools/sentinel1/rtc_metadata.py:
```diff
@@ -135,9 +136,10 @@ def platform(self) -> Optional[str]:
-    @property
-    def proj_bbox(self) -> Optional[str]:
-        return self.metadata['PROJ_BBOX']
+    # NOTE: redundent with projection.transform
+    # @property
+    # def proj_bbox(self) -> Optional[str]:
+    #    return self.metadata['PROJ_BBOX']
```
Since it's redundant, can we remove it (unless there's some reason to keep it in but commented)? Same w/ https://github.com/stactools-packages/sentinel1/pull/2/files#diff-68f6a98ce4447e3244540fbae6c5c887bde637ea0e4355be054f44bf8313aca4R33.

gadomski commented on src/stactools/sentinel1/rtc_metadata.py:
```diff
@@ -147,11 +149,14 @@ def epsg(self) -> Optional[str]:
-            'sentinel:grid_square': self.metadata['TILE_ID'][3:],
             'sentinel:product_ids': self.grd_ids,
+            # NOTE: change to 'valid_percent' in raster extension?
+            # https://github.com/stac-extensions/raster#statistics-object
```
Seems like it'd be worth using the `raster` extension's attribute, since it's more widely used than `sentinel`? What's the downside of using `raster`?

gadomski commented on src/stactools/sentinel1/stac.py:
```diff
@@ -20,22 +25,36 @@ def create_collection() -> pystac.Collection:
+        summaries=Summaries(summary_dict),
     )
 
+    # Fails validation i think b/c 'about' is not a recognized reltype
```
I don't see an enum in the `reltype` definition in https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/item.json. Can you post the validation failure?

scottyhq replied:
not pursuing this since i can put this information under 'providers' with the processing stac extension

gadomski commented on tests/test_commands.py:
```diff
@@ -3,19 +3,62 @@
+            collection.save(catalog_type=pystac.CatalogType.RELATIVE_PUBLISHED)
+            collection = pystac.read_file(
+                os.path.join(tmp_dir, 'collection.json'))
+            collection.validate()
```
Can you `validate_all` to check the items too?

gadomski reviewed (changes requested):
Thanks for the updates! I think stac-utils/pystac#479 might take a bit to resolve, so if it is a blocker I think we can merge this once changes are made and then open a new issue to add the processing extension later. A couple top-level notes:

> Looks like I can't request reviewers and don't have merge rights here

I gave you write access to the repo. I haven't set up branch protection yet, we can do that after the v0.1.0 release unless you want to do it now.

> Version is currently set to 0.1.5 b/c previously it matched stactools. I guess the first release of this subpackage should just be 0.1.0 though?

Yup, if it's not on PyPI yet, I'd go `v0.1.0`. We don't have a `stactools-packages` version policy and, IMO, we don't need one, since one of the goals of this reorganization was to allow packages to move at their own cadence. There _could_ be an argument for trying to track the stactools version that your package is based upon, but for now i think `v0.1` is good.

scottyhq commented:
Thanks so much @gadomski for the feedback! I think I addressed all your comments. Also added an examples/ folder and updated the CI to match what is currently in the template. And tests are passing!

So that this doesn't turn into an endless PR (I can't even remember when I started this back in the original stactools repo :) I'd like to merge, but I'd appreciate a final thumbs up from @gadomski, @matthewhanson, or @lossyrob.

I think it makes sense to wait on a first release and publishing to pypi until after stactools 0.2 is released. That also gives some time for additional changes.

My hope is that others can get involved going forward, perhaps adding additional S1 datasets such as GRD and SLC.

gadomski commented on README.md:
```diff
@@ -1,15 +1,68 @@
+  - [mgrs](https://github.com/stac-extensions/mgrs)
+  - [processing](https://github.com/stac-extensions/processing)
+
+A short description of the package and its usage.
```
Can probably nuke this boilerplate line.

gadomski commented on src/stactools/sentinel1/stac.py:
```diff
@@ -4,7 +4,12 @@
+from pystac.extensions.sat import OrbitState, SatExtension
+from pystac.extensions.sar import SarExtension
+from pystac.extensions.raster import RasterExtension, RasterBand
+# from pystac import Link
```
Can you just remove this line rather than commenting it out?

gadomski reviewed (changes requested):
Just two small nits but otherwise looks good.
